lib/signal-handler.js

```diff
@@ -0,0 +1,21 @@
+'use strict';
+
+var QEmitter = require('qemitter'),
+    signalHandler = new QEmitter();
+
+module.exports = signalHandler;
+
+process.on('SIGHUP', notifyAndExit(1));
+process.on('SIGINT', notifyAndExit(2));
```

Comment:
Not related to this PR, but I think it might be a good idea to treat second SIGINT in a row as "exit right now not waiting for anything". Sometimes in case of bad connection or grid problems in can take several minutes.